File: src/LIT.Smabu.UseCases/Catalogs/CatalogGroupDTO.cs
using LIT.Smabu.Domain.CatalogAggregate;
using LIT.Smabu.UseCases.Base;

namespace LIT.Smabu.UseCases.Catalogs
{
    public record CatalogGroupDTO(CatalogGroupId Id, CatalogId CatalogId, string DisplayName, string Name, string Description, CatalogItemDTO[] Items) : IDTO
    {
        internal static CatalogGroupDTO Create(CatalogGroup catalogGroup)
        {
            CatalogItemDTO[] items = catalogGroup.Items.Select(x => CatalogItemDTO.Create(x, catalogGroup)).ToArray();
            return new CatalogGroupDTO(catalogGroup.Id, catalogGroup.CatalogId, catalogGroup.DisplayName, catalogGroup.Name, catalogGroup.Description, items);
        }
    }
}

File: src/lit.smabu.react/src/types/domain/catalog-group-dto.ts

import { CatalogGroupId } from './catalog-group-id';
import { CatalogId } from './catalog-id';
import { CatalogItemDTO } from './catalog-item-dto';

export interface CatalogGroupDTO {
  id?: CatalogGroupId;
  catalogId?: CatalogId;
  name?: string | null;
  description?: string | null;
  items?: Array<CatalogItemDTO> | null;
  displayName?: string | null;
}
